File: Tutoria1/Tutoria1/DelegateTableView.m
//
//  DelegateTableView.m
//  Tutoria1
//
//  Created by Nicolás Gebauer on 20-03-15.
//  Copyright (c) 2015 Nicolás Gebauer. All rights reserved.
//

#import "DelegateTableView.h"
#import "Celda.h"

@implementation DelegateTableView

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return 10;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    Celda* celda = [tableView dequeueReusableCellWithIdentifier:@"IdentificadorTabla"];
    
    celda.labelVariable.text = [NSString stringWithFormat:@"Etiqueta numero %ld", indexPath.row];
    
    return celda;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    _numeroCelda = indexPath.row;
    [_referenciaViewController performSegueWithIdentifier:@"Transicion1" sender:_referenciaViewController];
}


@end

File: IoP App/IoP App/DelegateTableView.swift
//
//  DelegateTableView.swift
//  IoP App
//
//  Created by Nicolás Gebauer on 01-04-15.
//  Copyright (c) 2015 Nicolás Gebauer. All rights reserved.
//

import Foundation
import UIKit

class DelegateTableView : NSObject, UITableViewDelegate, UITableViewDataSource {
    var listaAlumnos : [String] = []
    weak var referenciaAlControlador : UIViewController?
    var contactoMostrar : Contacto?
    var listaContactos : ListaContactos?
    var labelDetallesLandscape : UILabel?
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        if let lista = listaContactos?.arrayContactos{
            return lista.count
        }
        else{
            return 0
        }
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        if let contacto = listaContactos?.arrayContactos[indexPath.row] {
            if contacto.celdaLetra {
                let celdaLetra = tableView.dequeueReusableCellWithIdentifier("IdentificadorCeldaLetra") as! CeldaLetra
                celdaLetra.LabelLetra.text = " " + contacto.nombreCompleto
                celdaLetra.LabelLetra.backgroundColor = UIColor.lightGrayColor()
                return celdaLetra
            }
            else if !contacto.celdaLetra {
                let celdaNombre = tableView.dequeueReusableCellWithIdentifier("IdentificadorCeldaNombre") as! CeldaNombre
                celdaNombre.LabelNombre.text = contacto.nombreCompleto
                return celdaNombre
            }
            else {
                return UITableViewCell()
            }
        }
        return UITableViewCell()
    }
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        if let referencia = referenciaAlControlador{
            if let contacto = listaContactos?.arrayContactos[indexPath.row] {
                contactoMostrar = contacto
                if referenciaAlControlador?.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClass.Compact && referenciaAlControlador?.traitCollection.verticalSizeClass == UIUserInterfaceSizeClass.Compact {
                    labelDetallesLandscape?.text = contacto.nombreCompleto
                }
                else if !contacto.celdaLetra {
                    referencia.performSegueWithIdentifier("IdentificadorTransicionVistaListaADetalle", sender: referencia)
                }
            }
        }
        tableView.deselectRowAtIndexPath(indexPath, animated: false)
    }
    
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        if let contacto = listaContactos?.arrayContactos[indexPath.row]{
            if contacto.celdaLetra {
                return 20
            }
            else if !contacto.celdaLetra {
                return 40
            }
        }
        return 0
    }
    
    func agregarContacto(nuevoContacto:Contacto){
        //agregar contactos
        println("Se agrego un nuevo contacto")
        println("Nombre: " + nuevoContacto.nombreCompleto)
        listaContactos?.agregarContacto(nuevoContacto)
    }
}